Run accumulo checks at WARN, and fix some easier spots

alerman opened the pull request.

alerman committed:
run accumulo import checks

alerman committed:
Clean up some of the references to accumulo code

alerman committed:
Adding a few more easy ones

alerman committed:
Update missed reference to accumulo internal code

ddanielr commented on warehouse/core/src/main/java/datawave/mr/bulk/BulkInputFormat.java:
```diff
@@ -54,8 +56,6 @@
 import org.apache.accumulo.core.security.TablePermission;
 import org.apache.accumulo.core.singletons.SingletonReservation;
 import org.apache.accumulo.core.util.Pair;
```
Pair can be replaced with [AbstractMap.SimpleEntry](https://docs.oracle.com/javase/8/docs/api/java/util/AbstractMap.SimpleEntry.html) at a later date.

ddanielr commented on checkstyle.xml:
```diff
@@ -2,13 +2,13 @@
+    <module name="ImportControl">
+        <!--Allows per-package enforcement for importing Accumulo 2.1.x public APIs -->
+        <property name="file" value="import-control-accumulo.xml"/>
+        <property name="severity" value="warning"/>
```
With this change, the build now shows these messages for disallowed imports
```
[WARN] /data/workspace/datawave/warehouse/query-core/src/main/java/datawave/core/iterators/ColumnRangeIterator.java:19:1: Disallowed import - org.apache.accumulo.core.iteratorsImpl.system.InterruptibleIterator. [ImportControl]
```

Verified that switching this severity line to `error` fails the build on ImportControl violations.

ddanielr reviewed (dismissed):
Looks good! 
The build prints out the disallowed imports without failing so they can be worked off incrementally without causing disruption.

ddanielr commented:
One thing to consider. This change fails if dropping down to a sub-directory to build via `mvn org.apache.maven.plugins:maven-checkstyle-plugin:3.3.0:check` and the documented `mvn -Pdocker,dist -DskipMicroservices clean install -T 1C` build command. 

Build failure due to checkstyle not finding the `import-control-accumulo.xml` file.  
Solution could be to add additional files per sub-directory, or figure out how to inject a root project path in for the file location. 

Approved since the README does not direct users to build from the sub-directories.

ddanielr committed:
Add root location prefix to fix sub-dir builds

Resolves the proper base directory for the import-control file so builds
from sub-directories will still work.

ddanielr commented:
> One thing to consider. This change fails if dropping down to a sub-directory to build via `mvn org.apache.maven.plugins:maven-checkstyle-plugin:3.3.0:check` and the documented `mvn -Pdocker,dist -DskipMicroservices clean install -T 1C` build command.
> 
> Build failure due to checkstyle not finding the `import-control-accumulo.xml` file. Solution could be to add additional files per sub-directory, or figure out how to inject a root project path in for the file location.
> 
> Approved since the README does not direct users to build from the sub-directories.

Added a commit that fixes this problem
